packages/clerk-js/src/ui/signUp/SignUpStart.tsx

```diff
@@ -302,7 +302,12 @@ function _SignUpStart(): JSX.Element {
       </Control>
     ) : null;
 
-  const atLeastOneFormField = nameField || usernameField || emailAddressField || phoneNumberField || passwordField;
+  const atLeastOneAuthFactor =
+    userSettings.attributes.email_address.used_for_first_factor ||
+    userSettings.attributes.phone_number.used_for_first_factor ||
+    (userSettings.attributes.password.required && userSettings.attributes.username.required);
+
+  const showFormFields = atLeastOneAuthFactor || !oauthOptions.length;
```

Comment:
🔧 @npetridis we should add another check for Web3 options. They behave in the exact same way as oauth options.